internal/outputgenerators/bonnmotion.go

```diff
@@ -114,36 +110,48 @@ func (b Bonnmotion) execute(command []string) error {
 	return err
 }
 
-// Calls BonnMotion to generate the Random Waypoint data for a given NodeGroup inside an Experiment.
-func (b Bonnmotion) generateRandomWaypointNodeGroup(exp experiment.Experiment, nodeGroup experiment.NodeGroup) {
-	logger.Trace("Generating Random Waypoint movements")
+// Calls BonnMotion to generate the for a given NodeGroup inside an Experiment.
+func (b Bonnmotion) generateNodeGroup(exp experiment.Experiment, nodeGroup experiment.NodeGroup) error {
+	logger.Tracef("Generating \"%s\" movements", nodeGroup.MovementModel.String())
+	var movementModelParameters []string
+	switch movementModel := nodeGroup.MovementModel.(type) {
```

Comment:
please declare movementModel outside of the switch for a better readabillity

Reply:
This has technical reasons. movementModel is only of the correct type (which is switched over), if declared inline in the switch statement.